controllers/discordactions.js

```diff
@@ -200,35 +201,63 @@ const updateDiscordNicknames = async (req, res) => {
       });
     }
 
-    const discordServerUsers = await dataAccess.retrieveDiscordUsers();
-    const nonSuperUsers = discordServerUsers.filter((user) => !user.roles.super_user);
+    const membersInDiscord = await getDiscordMembers();
+    const usersToBeEffected = [];
+    await Promise.all(
+      membersInDiscord.map(async (discordUser) => {
+        try {
+          const foundUserWithDiscordId = await fetchUser({ discordId: discordUser.user.id });
+
+          if (foundUserWithDiscordId.userExists) {
+            const isDeveloper = discordUser.roles.includes(discordDeveloperRoleId);
+            const isMaven = discordUser.roles.includes(discordMavenRoleId);
+            const isBot = discordUser.user.bot;
+            const isUsernameMatched = discordUser.nick === foundUserWithDiscordId.user.username;
+            const isSuperuser = foundUserWithDiscordId.user.roles.super_user;
+            if (isDeveloper && !isMaven && !isUsernameMatched && !isBot && !isSuperuser) {
+              usersToBeEffected.push({
+                discordId: foundUserWithDiscordId.user.discordId,
+                username: foundUserWithDiscordId.user.username,
+                first_name: foundUserWithDiscordId.user.first_name,
+              });
+            }
+          }
+        } catch (error) {
+          logger.error(`error getting user with matching discordId ${error.message}`);
+        }
+      })
+    );
 
-    const errorsArr = [];
-    let successCounter = 0;
-    let errorCounter = 0;
+    const totalNicknamesUpdated = { count: 0 };
+    const totalNicknamesNotUpdated = { count: 0, errors: [] };
 
     let counter = 0;
-    for (let i = 0; i < nonSuperUsers.length; i++) {
-      const { discordId, username } = nonSuperUsers[i];
+    for (let i = 0; i < usersToBeEffected.length; i++) {
+      const { discordId, username, first_name: firstName } = usersToBeEffected[i];
       try {
         if (counter % 10 === 0 && counter !== 0) {
-          await new Promise((resolve) => setTimeout(resolve, 4500));
+          await new Promise((resolve) => setTimeout(resolve, 3100));
+        }
+        if (!discordId) {
+          throw new Error("user not verified");
+        } else if (!username) {
+          throw new Error(`does not have a username`);
+        }
+        const response = await setUserDiscordNickname(username.toLowerCase(), discordId);
+        if (response.message) {
```

Comment:
response variable is of type Promise, so it may not have a message property yet. If the response.message property is not defined, the if statement will evaluate to false, and the code inside the block will not be executed. This means that the counter and totalNicknamesUpdated.count variables will not be incremented, even though the Discord nickname was successfully updated.

To fix this bug, you can use the following code:


if (response) {
  const message = await response.message;
  if (message) {
    counter++;
    totalNicknamesUpdated.count++;
  }
}